gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/token/impl/TokenServiceImpl.java

```diff
@@ -181,7 +191,12 @@ public Single<Token> refresh(String refreshToken, TokenRequest tokenRequest, Cli
                     // else, refresh token is used only once
                     return refreshTokenRepository.delete(refreshToken1.getValue())
                             .andThen(Single.just(refreshToken1));
-                });
+                })
+                .doOnEvent((token, error) -> auditService.report(AuditBuilder.builder(ClientTokenAuditBuilder.class)
+                                .token(TokenTypeHint.REFRESH_TOKEN, token != null ? token.getValue() : null)
+                                .tokenTarget(client)
+                                .revoked()
```

Comment:
**though**: should we add a revokation reason as here refresh toekn is revoked because it is used to generate a new access_token. :thinking: